core-web/apps/dotcms-ui/src/app/portlets/dot-containers/container-list/container-list.component.ts

```diff
@@ -1,8 +1,225 @@
-import { Component } from '@angular/core';
+import { Component, OnDestroy, OnInit, ViewChild } from '@angular/core';
+import { DotListingDataTableComponent } from '@components/dot-listing-data-table/dot-listing-data-table.component';
+import { MenuItem } from 'primeng/api';
+import { pluck, takeUntil } from 'rxjs/operators';
+import { Observable, Subject } from 'rxjs';
+import { DotMessageService } from '@services/dot-message/dot-messages.service';
+import { DotAlertConfirmService } from '@services/dot-alert-confirm';
+import { DotPushPublishDialogService } from '@dotcms/dotcms-js';
+import { DotRouterService } from '@services/dot-router/dot-router.service';
+import { ActionHeaderOptions } from '@models/action-header';
+import { DataTableColumn } from '@models/data-table';
+import { DotContainer } from '@models/container/dot-container.model';
+import { DotContentState } from '@dotcms/dotcms-models';
+import {
+    DotContainerListState,
+    DotContainerListStore
+} from '@portlets/dot-containers/container-list/store/dot-container-list.store';
 
 @Component({
     selector: 'dot-container-list',
     templateUrl: './container-list.component.html',
-    styleUrls: ['./container-list.component.scss']
+    styleUrls: ['./container-list.component.scss'],
+    providers: [DotContainerListStore]
 })
-export class ContainerListComponent {}
+export class ContainerListComponent implements OnInit, OnDestroy {
+    @ViewChild('listing', { static: false })
+    vm$ = this.store.vm$;
+
+    listing: DotListingDataTableComponent;
+    selectedContainers: DotContainer[] = [];
+
+    containerBulkActions$: Observable<MenuItem[]>;
+    tableColumns$: Observable<DataTableColumn[]>;
+    actionHeaderOptions$: Observable<ActionHeaderOptions>;
+    addToBundleIdentifier$: Observable<string>;
+
+    updateBundleIdentifier = this.store.updateBundleIdentifier;
+
+    private isEnterPrise: boolean;
+    private hasEnvironments: boolean;
+    private destroy$: Subject<boolean> = new Subject<boolean>();
+
+    constructor(
+        private store: DotContainerListStore,
+        private dotMessageService: DotMessageService,
+        private dotAlertConfirmService: DotAlertConfirmService,
+        private dotPushPublishDialogService: DotPushPublishDialogService,
+        private dotRouterService: DotRouterService
+    ) {}
+
+    ngOnInit(): void {
+        this.store.updateTableColumns(this.setContainerColumns());
+        this.store.updateContainerBulkActions(this.setContainerBulkActions());
```

Comment:
Don't update the state twice because it will trigger the change detection twice.

Everything you are doing here, which is unit the state, can be done in the store itself as far as I can see.

The `setContainerColumns` and `setContainerBulkActions` can be done in the store.

More info: https://ngrx.io/guide/component-store/initialization#lazy-initialization